docs/csharp/how-to/search-strings.md

```diff
@@ -0,0 +1,66 @@
+ms.author: "wiwagn"
+---
+
+# How to: search strings
+
+You have two main strategies that you can use to search for text in strings. Member methods of the <xref:System.String> class search for specific text. Regular expressions search for patterns in text.
+
+[!INCLUDE[interactive-note](~/includes/csharp-interactive-note.md)]
+
+The [string](../language-reference/keywords/string.md) type, which is an alias for the <xref:System.String?displayProperty=nameWithType> class, provides a number of useful methods for searching the contents of a string. Among them are <xref:System.String.Contains%2A>, <xref:System.String.StartsWith%2A>, <xref:System.String.EndsWith%2A>, <xref:System.String.IndexOf%2A>, <xref:System.String.LastIndexOf%2A>. The <xref:System.Text.RegularExpressions.Regex?displayProperty=nameWithType> class provides a rich vocabulary to search for patterns in text. In this article, you learn these techniques and how to choose the best method for your needs.
+
+## Does a string contain text?
+
+The <xref:System.String.Contains%2A?displayProperty=nameWithType>, <xref:System.String.StartsWith%2A?displayProperty=nameWithType> and <xref:System.String.EndsWith%2A?displayProperty=nameWithType> methods search a string for specific text. The following example shows each of these methods, and a variation that uses a case insensitive search:
+
+[!code-csharp-interactive[search strings using methods](../../../samples/snippets/csharp/how-to/strings/SearchStrings.cs#1)]
+
+The preceding example demonstrates an important point for using these methods. Searches are **case-sensitive** by default. You use the <xref:System.StringComparison.CurrentCultureIgnoreCase> enum value to specify a case insensitive search.
+
+## Where does the sought text occur in a string?
+
+The <xref:System.String.IndexOf%2A> and <xref:System.String.LastIndexOf%2A> methods also search for text in strings. These methods return the location of the text being sought. If the text isn't found, they return `-1`. The following example shows a search for the first and last occurrence of the word "methods", and displays the text in between.
+  
+[!code-csharp-interactive[search strings for indices](../../../samples/snippets/csharp/how-to/strings/SearchStrings.cs#2)]
+
+## Finding specific text using regular expressions
+
+The <xref:System.Text.RegularExpressions.Regex?displayProperty=nameWithType> class can be used to search strings. These searches can range in complexity from simple to complicated text patterns.
+
+The following code example searches for the word "code" in a sentence, ignoring case. The static method <xref:System.Text.RegularExpressions.Regex.IsMatch%2A?displayProperty=nameWithType> performs the search. You give it the string to search and a search pattern. In this case, a third argument specifies case-insensitive search. For more information, see <xref:System.Text.RegularExpressions.RegexOptions?displayProperty=nameWithType>.  
```

Comment:
Rather than present an example that can better be done with a string method (so we're illustrating a worst practice), I'd do one that actually uses a pattern. Possibly a simple search, such as a one that uses the pattern `the(ir)?` to match either "the" or "their"

Reply:
Or:
- search for a noun in both singular and plural form
- search for both '2' and 'two'

Reply:
nit: if you keep the existing example, you should change the description to note that it search for the first occurrence of the word "code" in a sentence.